src/google/protobuf/descriptor.proto

```diff
@@ -351,6 +351,14 @@ message FileOptions {
   // generated classes from this .proto. There is no default.
   optional string objc_class_prefix = 36;
 
+  // Namespace for generated classes; defaults to the package.
+  optional string csharp_namespace = 37;
+
+  // A nested namespace for the umbrella class.  Helpful for name collisions caused by
```

Comment:
We previously had an "umbrella class name" option, which explained this comment a bit more clearly, I think. As it is, it's not obvious what the umbrella class is - and I'm struggling to work out exactly what this namespace is vs the csharp_namespace.